src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -556,6 +562,10 @@ object FunctionRegistry {
     def typ = TTBoxed
   }
 
+  private def nonceToNullable[T : TypeInfo, U >: Null](check: Code[T] => Code[Boolean], v: Code[T], ifPresent: Code[T] => Code[U]): CM[Code[U]] = for (
```

Comment:
I'm referring to [nonce words](https://en.wikipedia.org/wiki/Nonce_word), which is supposed to be meaningless outside of some context for which it was created. Here, the "nonce" identifies the null-case. `check` returns true when its argument is the nonce value.